### ISSUE TITLE
Should we overstrictly typecheck object arguments

### ISSUE THREAD

**Issue by bogeeee** [bogeeee/restfuncs#1]
Let's say we have a service with runtime typechecking (shielding) enabled _(as of > 0.9.0 release)_ :

```
type User = {
    name: string,
    lastName?: string,        
}

class MyService {
   changeMyUserSettings(user: User) {
   }
}
```

What if an attacker calls it with
```

const poisoned = {
   name: "Hans", 
   bpms_permissions: { allowWrite: "all" }
}

myRemoteService.changeMyUserSettings(poisoned);
```

Currently you can "poison" it with any additionaly fields, i.e. bpms_permissions and maybe some deeply nested library will evaluate it somehow or it gets saved in the database. Currently you have to just have some knowledge, how pure your code behind the service is and be very carefull, when allowing object parametes that get passed on to other API. So, what do you think ? Is this really an issue in practice ? Should we overstrictly forbid the use of additional properties and what would be the implication. Should we add some decorators for this like @tsUnlistedFields("erase"|"allow"|"error"). Is this even possible to implement and not clash mit Record / Indexed / Union types ?

**Comment by bogeeee** [comment]
Just forget this issuse. Typescript marks this as an error and i just missthought it / found a bug.  0.9 is not yet released / it will be fixed then.

**Comment by bogeeee** [comment]
[See also](https://github.com/typescript-rtti/typescript-rtti/issues/92)

**Comment by bogeeee** [comment]
Fixed in 1d3a156f. Extra properties are not allowed. In the future, we could auto remove them via @remote({shapeArguments:true})
